### ISSUE TITLE
Consolidate Block size dependant variables

### ISSUE THREAD

**Issue by Naviabheeman** [chaintope/tapyrus-core#236]
Tapyrus-core contains many statically defined constants like MAX_BLOCK_SIZE, DEFAULT_MAX_BLOCK_SIZE etc. In #234 we made Block size dynamic using xfield. There are many more constants derived from this like

'''
  static const size_t MAX_OUTPUTS_PER_BLOCK = MAX_BLOCK_SIZE / GetSerializeSize(CTxOut(), SER_NETWORK, PROTOCOL_VERSION);
  static const size_t MAX_INPUTS_PER_BLOCK = MAX_BLOCK_SIZE / GetSerializeSize(CTxIn(), SER_NETWORK, PROTOCOL_VERSION);
  static const int64_t MAX_BLOCK_SIGOPS = MAX_BLOCK_SIZE/50;
  static const unsigned int MAX_STANDARD_TX_SIGOPS = MAX_BLOCK_SIGOPS/5; 
  block transactions count check - MAX_BLOCK_SIZE / MIN_TRANSACTION_SIZE & MAX_BLOCK_SIZE / MIN_SERIALIZABLE_TRANSACTION_SIZE
'''
When max block size increases these have to increase too. 

We need to add one more structure in Federation Params to store all these derived limits. When max block size is changed using xfield in a block these limits will change accordingly. All newer blocks are validated using the new limits.

**To Check:**

P2P layer buffer sizes are changed to match MAX_BLOCK_SIZE. Check that it doesn't cause problems in all scenarios like load block file, block chain reorg, reindex, reindex chainstate, serving historical blocks etc.

**Comment by Naviabheeman** [comment]
Transaction size is limited by multiple constants:

**static const size_t MIN_TRANSACTION_SIZE = 60;** 
// 60 is the lower bound for the size of a valid serialized CTransaction
**static const size_t MIN_SERIALIZABLE_TRANSACTION_SIZE = 10;** 
// 10 is the lower bound for the size of a serialized CTransaction
/** The maximum size for transactions we're willing to relay/mine */
**static const unsigned int MAX_STANDARD_TX_SIZE = 100000;**
/** The minimum non-witness size for transactions we're willing to relay/mine (1 segwit input + 1 P2WPKH output = 82 bytes) */
**static const unsigned int MIN_STANDARD_TX_SIZE = 82;**

In Tapyrus non-standard transactions are not accepted anymore. We need to make the constant more accurate.

| part  | min size  | p2pkh | p2sh | cp2pkh | cp2sh | 
|---|:----|:---|:---|:---|:---|
| vin  | 4(len) + 32 + 4 + ____ + 4 | 64 | 64 |  64 |  64 |
| vout  | 4(len) + 4 + _____ | 67 | 25 |  111 |  59 |
| nFeatures  | 4 |
| nLockTime  | 4 |
| Total  | min = 60 | 60 + 131 | 60 + 89 | 60 + 175 | 60 + 123 |



And checked in multiple places:

_AcceptToMemoryPoolWorker:_
```
    if (::GetSerializeSize(tx, SER_NETWORK, PROTOCOL_VERSION | SERIALIZE_TRANSACTION_NO_WITNESS) < MIN_STANDARD_TX_SIZE)
        return state.DoS(0, false, REJECT_NONSTANDARD, "tx-size-small");
```

_CheckTransaction:_
```

    if (::GetSerializeSize(tx, SER_NETWORK, PROTOCOL_VERSION | SERIALIZE_TRANSACTION_NO_WITNESS) > MAX_BLOCK_SIZE)
        return state.DoS(100, false, REJECT_INVALID, "bad-txns-oversize");
```

_IsStandardTx:_
```
    unsigned int sz = GetSerializeSize(tx, SER_NETWORK, PROTOCOL_VERSION);
    if (sz > MAX_STANDARD_TX_SIZE) {
        reason = "tx-size";
        return false;
    }
```

_CWallet::CreateTransaction:_
```
        // Limit size
        if (GetSerializeSize(tx, SER_NETWORK, PROTOCOL_VERSION) > MAX_STANDARD_TX_SIZE)
        {
            strFailReason = _("Transaction too large");
            return false;
        }
```

**Comment by Naviabheeman** [comment]
considering in more detail, MAX_OUTPUTS_PER_BLOCK is used exactly once in the whole code base, in AccessByTxid. when we lookup the level db coin cache, we need outpoint i.e txid and vin. In AccessByTxid we try to find if any unspent output is available using every combination from (txid, 0), (txid, 1)… (txid, MAX_OUTPUTS_PER_BLOCK). 
It would be so much more efficient to  implement this in the DB wrapper. in fact there is an example in the db_tests (std::string AllEntriesFor(const Slice& user_key))

similarly, MAX_INPUTS_PER_BLOCK is used exactly once in Unserialize of CTxUndo class. The limit check seems unnecessary. Deserialize of undo information is used only during a reorg when the block undo info is read from the undo block file.  Do we really need to check the limit here? But it has been there from when it was implemented in commit cb2c7fdac2dc74368ed24ae4717ed72178956b92.

MAX_BLOCK_SIZE / MIN_TRANSACTION_SIZE and MAX_BLOCK_SIZE / MIN_SERIALIZABLE_TRANSACTION_SIZE checks can be reimplemented using the dynamic current block size.

In conclusion only MAX_BLOCK_SIGOPS and MAX_STANDARD_TX_SIGOPS are important to be stored in Federation params. They need to change based on block size. All other constants can be removed or reimplemented to be independent of block size.

**Comment by Naviabheeman** [comment]
implemented in #245

MAX_OUTPUTS_PER_BLOCK and MAX_INPUTS_PER_BLOCK are unchanged
